tests/js/core/widget/core/Appbar/Appbar.js

```diff
@@ -0,0 +1,246 @@
+				assert.ok(true, "_setTitlesOpacity has been called");
+			});
+
+			widget._onDrag(testEvent);
+
+			assert.equal(widget.element.style.height, "100px", "_onDrag: element height has been changed");
+		});
+
+		test("Test of _onDragStart appbar", 4, function (assert) {
+			var widget = new Appbar(),
+				element = document.getElementById("appbar-1"),
+				testEvent = {
+					detail: {
+						direction: "down",
+						deltaY: 20
+					}
+				};
+
+			widget.element = element;
+			widget.options.expandingEnabled = true;
+			widget._dragStartingHeight = 80;
+			widget._lockExpanding = false;
+			widget._appbarState = "COLLAPSED";
+			self._scrolledToTop = true;
+			widget._currentHeight = 80;
+			widget._expandedHeight = 100;
+			element.style.height = "100px";
+			widget._ui = {
+				expandedTitleContainer: {
+					children: [{
+						offsetHeight: 100
+					}, {
+						offsetHeight: 100
+					}]
+				}
+			}
+
+			widget._onDragStart(testEvent);
+
+			assert.equal(widget._appbarState, "DRAGGING", "_appbar state is set to dragging");
+			assert.ok(widget.element.classList.contains("ui-appbar-dragging"), "element contains dragging class");
+			assert.equal(widget._dragStartingHeight, 100, "dragStartingHeight has correct value");
+			assert.equal(widget._expandedTitleHeight, 200, "_expandedTitleHeight has been calculated correctly");
+		});
+
+		test("Test of collapse appbar", 6, function (assert) {
+			var widget = new Appbar(),
+				element = document.getElementById("appbar-1");
+
+			widget.element = element;
+			widget._appbarState = "";
+
+			helpers.stub(widget, "_setTitlesOpacity", function (value) {
+				assert.ok(true, "_setTitlesOpacity has been called");
+				assert.equal(value, 0, "Oppacity of expanded title is set to 0");
+			});
+
+			widget.collapse();
+
+			assert.equal(widget.element.style.height, "", "collapse: element height has been changed");
+			assert.equal(widget._currentHeight, 0, "_currentHeight equals 0");
+			assert.ok(!widget.element.classList.contains("ui-appbar-expanded"), "element dosn't contain expanded class");
+			assert.equal(widget._appbarState, "COLLAPSED", "_appbar state id collapsed");
+		});
+
+		test("Test of expand appbar", 6, function (assert) {
+			var widget = new Appbar(),
+				element = document.getElementById("appbar-1");
+
+			widget.element = element;
+			widget._appbarState = "";
+
+			helpers.stub(widget, "_setTitlesOpacity", function (value) {
+				assert.ok(true, "_setTitlesOpacity has been called");
+				assert.equal(value, 1, "Oppacity of expanded title is set to 1");
+			});
+
+			widget.expand();
+
+			assert.equal(widget.element.style.height, "", "expand: element height has been changed");
+			assert.equal(widget._currentHeight, 0, "_currentHeight equals 0");
+			assert.ok(widget.element.classList.contains("ui-appbar-expanded"), "element contains expanded class");
+			assert.equal(widget._appbarState, "EXPANDED", "_appbar state id collapsed");
```

Comment:
Need to change the description as "_appbar state is expaned"

Reply:
Done.